extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/BasicImpl.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.vertx.http.runtime.security;
+
+import java.util.Optional;
+
+import io.quarkus.vertx.http.security.event.Basic;
+
+final class BasicImpl implements Basic {
```

Comment:
Michal, should we make it more consistent with https://quarkus.io/guides/security-openid-connect-multitenancy#programmatic-startup, as far using the builder is concerned ?

`Basic` interface would probably have `enable()` shortcut which would produce a `BasicBuilder`, and `realm` shortcut that would be doc-ed to also enable it, as well as a plain `bulder()`, something like that... But users are calling `.build()` and then the runtime collecting the instance somehow...

That should also align well with configuring `basic/form` without using events where users would call `build()` I guess.

Thoughts ?

Reply:
> realm shortcut that would be doc-ed to also enable it

I actually had it in previous versions, but I dropped it because I realized that it could technically be that some code (or even an extension) configures the basic for users, but doesn't want to enable the mechanism. I think it's not important scenario, so let's do it (enable the basic with "realm")

Reply:
Hi @michalvavrik Yeah, on one hand having `Basic.create(basic.builder()....build())` looks a bit noisy but then we would be able to do, outside of the event. `httpSecurity.path("basic").basic(Basic.realm("TestRealm").build())` etc (though I'd advocate for the actual config be done before path here, and refer to whatever is configured with `.path(...).basic()` which we can discuss later in any case)

Thanks